Yosemite/Yosemite/Stores/ProductStore.swift

```diff
@@ -531,20 +535,41 @@ private extension ProductStore {
         }
     }
 
+    func identifyLanguage(siteID: Int64,
+                          string: String,
+                          feature: GenerativeContentRemoteFeature,
+                          completion: @escaping (Result<String, Error>) -> Void) {
+        Task {
+            let result = await Result {
+                try await generativeContentRemote.identifyLanguage(siteID: siteID,
+                                                                   string: string,
+                                                                   feature: feature)
+            }
+            await MainActor.run {
+                completion(result)
+            }
+        }
+    }
+
     func generateProductDescription(siteID: Int64,
                                     name: String,
                                     features: String,
+                                    language: String,
                                     completion: @escaping (Result<String, Error>) -> Void) {
-        let prompt = [
-            "Write a description for a product with title ```\(name)``` and features: ```\(features)```.",
-            "Identify the language used in the product title and features and use the same language in your response.",
-            "Make the description 50-60 words or less.",
-            "Use a 9th grade reading level.",
-            "Perform in-depth keyword research relating to the product in the same language of the product title, " +
-            "and use them in your sentences without listing them out."
-        ].joined(separator: "\n")
```

Comment:
❓ Just curious: is there a reason why we should move the prompt inside of `Task`?

Reply:
I changed this when we were sending the request for language identification from this task. 

I moved the prompts outside in https://github.com/woocommerce/woocommerce-ios/pull/10169/commits/3154d66f96f49217bedf30e5e3382c52dc3143a1